npc/repairman/repairman.gd

```diff
@@ -0,0 +1,40 @@
+class_name Repairman
+extends Node2D
+
+var repair_amount: int = 1
+var repair_delay: float = 1.0
+
+var _health: Health
+var _is_repairing: bool = false
+
+
+func _init(health: Health) -> void:
+	print("[Repairman] Created")
+	_health = health
+	_health.took_damage.connect(_on_damage_taken)
+
+	act()
+
+
+func act() -> void:
+	if _health.is_full():
+		print("[Repairman] Health is full, stopping repair")
+		_is_repairing = false
+		return
+
+	_is_repairing = true
+
+	await repair()
+
+	act()
+
+
+func repair() -> void:
+	print("[Repairman] Repairing")
+	_health.heal(repair_amount)
+	await get_tree().create_timer(repair_delay).timeout
```

Comment:
not sure if it's worth it but seeing this same method sequence quite a bit. maybe we could have a utility `waitFor(seconds: float)` that wires everything up and awaits

Reply:
Ah yeah, I'm not sure if that works with `get_tree()` since it's a `self` method, but we can figure it out.

Also, I'm not 100% sure if it would be better to keep a timer allocated always for this stuff, but we'll see